llvm/lib/Analysis/IVDescriptors.cpp

```diff
@@ -741,10 +739,9 @@ RecurrenceDescriptor::isFindIVPattern(RecurKind Kind, Loop *TheLoop,
                                   : APInt::getMinValue(NumBits);
         ValidRange = ConstantRange::getNonEmpty(Sentinel + 1, Sentinel);
       } else {
-        assert(IsSigned && "Only FindFirstIV with SMax is supported currently");
-        ValidRange =
-            ConstantRange::getNonEmpty(APInt::getSignedMinValue(NumBits),
-                                       APInt::getSignedMaxValue(NumBits) - 1);
+        APInt Sentinel = IsSigned ? APInt::getSignedMaxValue(NumBits)
+                                  : APInt::getMaxValue(NumBits);
+        ValidRange = ConstantRange::getNonEmpty(Sentinel + 1, Sentinel);
```

Comment:
On second thought, I think  the wrapped range isn't sufficient and this will need to initialize the range with min & max - 1 explicitly as suggested earlier (and also clearer)

Currently test coverage is missing for the case where the IV range includes both signed and unsigned sentinel values.